psp-examples/multisig/src/multisigParams.ts

```diff
@@ -0,0 +1,270 @@
+import { BN, BorshAccountsCoder, utils } from "@coral-xyz/anchor";
+import { IDL } from "./types/multisig";
+import { Account, Utxo } from "@lightprotocol/zk.js";
+import nacl from "tweetnacl";
+import { MAX_SIGNERS } from "./constants";
+import { assert } from "chai";
+const bs58 = require("bs58");
+// add encrypt to primitive, nonces are H(base_nonce||pubkey), H(base_nonce||pubkey), H(base_nonce||pubkey), etc.
+// if only one recipient use nonce directly
+// fields [base_nonce], [encryptedAes1,..., encryptedAesN ], [aesCiphertext],
+// standardized, the first 32 bytes are the pubkey,
+// encrypt to aes
+
+export class MultisigParams {
+  signersEncryptionPublicKeys: Array<Uint8Array>;
+  threshold: BN;
+  publicKeyX: Array<Uint8Array>;
+  publicKeyY: Array<Uint8Array>;
+  poseidon: any;
+  nrSigners: BN;
+  appDataHash?: string;
+  seed: Uint8Array;
+  account: Account;
+  priorMultiSigSlot: BN;
+  priorMultiSigHash: Uint8Array;
+  priorMultiSigSeed: Uint8Array;
+
+  constructor({
+    poseidon,
+    threshold,
+    nrSigners,
+    publicKeyX,
+    publicKeyY,
+    signersEncryptionPublicKeys,
+    seed,
+    priorMultiSigSlot,
+    priorMultiSigHash,
+    priorMultiSigSeed,
+  }: {
+    poseidon: any;
+    threshold: number;
+    nrSigners: number;
+    publicKeyX: Array<Uint8Array>;
+    publicKeyY: Array<Uint8Array>;
+    signersEncryptionPublicKeys: Array<Uint8Array>;
+    seed: Uint8Array;
+    priorMultiSigSlot: BN;
+    priorMultiSigHash: Uint8Array;
+    priorMultiSigSeed: Uint8Array;
+  }) {
+    this.threshold = new BN(threshold);
+    this.publicKeyX = publicKeyX;
+    this.publicKeyY = publicKeyY;
+    this.nrSigners = new BN(nrSigners);
+    this.signersEncryptionPublicKeys = signersEncryptionPublicKeys;
+    this.appDataHash = MultisigParams.getHash(
+      poseidon,
+      MultisigParams.toArray(
+        poseidon,
+        threshold,
+        nrSigners,
+        publicKeyX,
+        publicKeyY,
+      ),
+    );
+    this.seed = seed;
+    this.account = new Account({ poseidon, seed: bs58.encode(seed) });
+    this.priorMultiSigSlot = priorMultiSigSlot;
+    this.priorMultiSigHash = priorMultiSigHash;
+    this.priorMultiSigSeed = priorMultiSigSeed;
+  }
+
+  static async createNewMultiSig({
+    poseidon,
+    signers,
+    threshold,
+    randomSeed = nacl.randomBytes(32),
+  }: {
+    poseidon: any;
+    signers: Account[];
+    threshold: number;
+    randomSeed?: Uint8Array;
+  }) {
+    if (signers.length > MAX_SIGNERS) {
+      throw new Error(`Too many signers ${signers.length} > 6`);
+    }
+    const nrSigners = signers.length;
```

Comment:
This variable is created here, but in the line up and down `signers.length` is still used directly. :stuck_out_tongue: 

Please either define that variable at the top and use consistently, or just remove it and always access the field. I have no hard preference for either of solutions (accessing struct fields is cheap, I don't mind repeating it).